src/query/storage/local/storage.go

```diff
@@ -72,26 +227,19 @@ func (s *localStorage) Fetch(ctx context.Context, query *storage.FetchQuery, opt
 	// cluster that can completely fulfill this range and then prefer the
 	// highest resolution (most fine grained) results.
 	// This needs to be optimized, however this is a start.
+	fanout, namespaces, err := s.resolveClusterNamespacesForQuery(query.Start, query.End)
```

Comment:
couple thoughts:
- maybe it'd be good to add a way for users to see what data source was hit for each series
- do you want to add tests for the various combinations - just ensuring we override to the right storage based on the heuristics in code